doc_search_api/formatters/doctors_formatter.py

```diff
@@ -1,16 +1,33 @@
 import json
+from doc_search_api.services.google_distance_service import GoogleDistanceService
 
 class DoctorsFormatter:
+    def __init__(self, coords):
+        self.coords = coords if coords else None
+
     def format(self, docs, location):
         response = []
+        coordinates = []
         for doc in docs:
             if doc.practice['location'] == location:
                 result = {}
                 result['practice'] = doc.practice
                 result['profile'] = doc.profile
                 response.append(result)
-        sorted_distance = sorted(
-            response,
-            key=lambda d: d['practice']['distance']
-        )
-        return json.dumps(sorted_distance)
+                coordinates.append(
+                    [doc.practice['lat'], doc.practice['lon']]
+                )
+
+        if self.coords:
+            distances = json.loads(GoogleDistanceService().distances(self.coords[0], self.coords[1], coordinates))
+
+            for index, doc in enumerate(response):
+                response[index]['practice']['distance'] = distances['rows'][0]['elements'][index]['distance']['text']
+
+            sorted_distance = sorted(
+                response,
+                key=lambda d: d['practice']['distance']
+            )
+
+            return json.dumps(sorted_distance)
+        return json.dumps(response)
```

Comment:
If coordinates are supplied in query, distances will be calculated in one batch to google api instead of per doctor.